File: services/expense-manager-service/app/use_cases/models/wealth.py
"""Use case models for Wealth/Portfolio analytics."""

from app.use_cases.models.base import BaseInput


class WealthSummary(BaseInput):
    """Model for high-level current wealth aggregation."""

    total_assets: float
    total_invested_assets: float
    total_returns_assets: float
    percentage_returns_assets: float
    total_liabilities: float
    total_original_liabilities: float
    total_repaid_liabilities: float
    accumulated_interest_liabilities: float
    net_worth: float


class HistoricalNetWorthPoint(BaseInput):
    """Model for monthly historical net worth timeline point."""

    date: str  # Format: YYYY-MM
    total_assets: float
    total_liabilities: float
    net_worth: float


class WealthCategoryAllocation(BaseInput):
    """Model for allocation breakdown by category."""

    category_name: str
    category_code: str
    total_value: float
    percentage: float


class WealthAllocation(BaseInput):
    """Model for complete portfolio allocation breakdown."""

    assets: list[WealthCategoryAllocation]
    liabilities: list[WealthCategoryAllocation]
    total_assets_value: float
    total_liabilities_value: float
    debt_to_asset_ratio: float
    liquidity_ratio: float
    equity_financed_pct: float
    liabilities_financed_pct: float
    leverage_status_label: str
    leverage_status_type: str
    liquidity_status_label: str
    liquidity_status_type: str

File: keys-personal-assist-ui/src/types/wealth.ts
export interface WealthSummary {
  totalAssets: number;
  totalInvestedAssets: number;
  totalReturnsAssets: number;
  percentageReturnsAssets: number;
  totalLiabilities: number;
  totalOriginalLiabilities: number;
  totalRepaidLiabilities: number;
  accumulatedInterestLiabilities: number;
  netWorth: number;
}

export interface HistoricalNetWorthPoint {
  date: string; // Format: YYYY-MM
  totalAssets: number;
  totalLiabilities: number;
  netWorth: number;
}

export interface WealthCategoryAllocation {
  categoryName: string;
  categoryCode: string;
  totalValue: number;
  percentage: number;
}

export interface WealthAllocation {
  assets: WealthCategoryAllocation[];
  liabilities: WealthCategoryAllocation[];
  totalAssetsValue: number;
  totalLiabilitiesValue: number;
  debtToAssetRatio: number;
  liquidityRatio: number;
  equityFinancedPct: number;
  liabilitiesFinancedPct: number;
  leverageStatusLabel: string;
  leverageStatusType: 'SUCCESS' | 'WARNING' | 'ERROR';
  liquidityStatusLabel: string;
  liquidityStatusType: 'SUCCESS' | 'WARNING';
}
